# Syllabus {-}

## Course Overview

Informatics and Statistics for Molecular Biologists (MOLB 7900) teaches students to design and analyze experiments commonly used in molecular biology. The course is organized around the Central Dogma (DNA > RNA > Protein) wherein each block presents 2-3 experimental approaches. Each week, a new experiment is introduced with a discussion of appropriate design and statistical considerations. The remaining weeks' classes are devoted to digging into the analysis of a sample data set.

The course begins with a 4 week "boot camp" designed to get students familiar with and bring them up to speed on using software for shell programming and data analysis with R and Python. We also establish accounts on Github for problem set submission.

### Block 0: Bootcamp

	A: Shell programming
	B: R Studio
	C: Python
	D: Integration with Git

### Block 1: DNA

	A: Chromatin accessibility
	B: Motif finding
	C: Variant Calling

### Block 2: RNA

  A: Quantitative PCR
	B: Bulk mRNA-seq
	C: RNA-protein interaction (CLIP)

### BLOCK 3: PROTEIN

	A: Mass spectrometry (counts)
	B:  
	C: Fluorescent protein localization (Chad)

## Course Objectives

- Cater to students of many backgrounds (more computational or more biological)
- Be able to formulate questions that are testable with computational techniques
- Understand the limitations of sequencing-based techniques
- Be fluent in statistical considerations for different approaches and design well-controlled experiments that can be analyzed using statistical tests
- Be fluent in command-line programming, scripting (Python) and data analysis / viz (R / R Studio)
- Understand the value of internet-based analysis tools (NCBI BLAST etc)
- Use reproducible software development approaches (Github) and dynamic documents (Rmarkdown)
- Independently conceive of and implement a soup to nuts reanalysis of an existing data set, which are presented to the class.

## Class Schedule

Monday, Wednesday, and Friday, 1-2:30pm from Sept 10 to Dec 10. 
